tests/js/blockPRNG.scrypttest.js

```diff
@@ -0,0 +1,185 @@
+
+
+        preimage = getPreimage(tx, blockchainPRNG.lockingScript, inputSatoshis)
+
+        const sig = signTx(tx, alicePrivateKey, blockchainPRNG.lockingScript, inputSatoshis);
+
+        const result = blockchainPRNG.bet(new BlockHeader({
+            version: new Bytes(uint322bin(header.version)),
+            prevBlockHash: new Sha256(toLittleIndian(header.previousblockhash)),
+            merkleRoot: new Sha256(toLittleIndian(header.merkleroot)),
+            time: new Bytes(uint322bin(header.time)),
+            bits: new Bytes(toLittleIndian(header.bits)),
+            nonce: new Bytes(uint322bin(header.nonce))
+        }), buildMerklePath(merklePath), sig, preimage).verify()
+
+        expect(result.success, result.error).to.be.true
+    });
+
+
+    it('blockchainPRNG should fail when using wrong block header', () => {
+
+        tx.addOutput(new bsv.Transaction.Output({
+            script: bsv.Script.buildPublicKeyHashOut(alicePrivateKey.toAddress()),
+            satoshis: outputAmount
+        }));
+
+
+        blockchainPRNG.txContext = { tx, inputIndex, inputSatoshis }
+
+
+        preimage = getPreimage(tx, blockchainPRNG.lockingScript, inputSatoshis)
+
+        const sig = signTx(tx, alicePrivateKey, blockchainPRNG.lockingScript, inputSatoshis);
+
+        const result = blockchainPRNG.bet(new BlockHeader({
+            version: new Bytes(uint322bin(header.version)),
+            prevBlockHash: new Sha256(toLittleIndian(header.previousblockhash)),
+            merkleRoot: new Sha256(toLittleIndian(header.merkleroot)),
+            time: new Bytes(uint322bin(header.time)),
+            bits: new Bytes(toLittleIndian(header.bits)),
+            nonce: new Bytes(uint322bin(header.nonce + 1))
+        }), buildMerklePath(merklePath), sig, preimage).verify()
+
+        expect(result.success, result.error).to.be.false
+    });
+
+
+    it('blockchainPRNG should fail when using wrong wrongMerklePath', () => {
+
+        tx.addOutput(new bsv.Transaction.Output({
+            script: bsv.Script.buildPublicKeyHashOut(alicePrivateKey.toAddress()),
+            satoshis: outputAmount
+        }));
+
+
+        blockchainPRNG.txContext = { tx, inputIndex, inputSatoshis }
+
+        preimage = getPreimage(tx, blockchainPRNG.lockingScript, inputSatoshis)
+
+        const sig = signTx(tx, alicePrivateKey, blockchainPRNG.lockingScript, inputSatoshis);
+
+        const result = blockchainPRNG.bet(new BlockHeader({
+            version: new Bytes(uint322bin(header.version)),
+            prevBlockHash: new Sha256(toLittleIndian(header.previousblockhash)),
+            merkleRoot: new Sha256(toLittleIndian(header.merkleroot)),
+            time: new Bytes(uint322bin(header.time)),
+            bits: new Bytes(toLittleIndian(header.bits)),
+            nonce: new Bytes(uint322bin(header.nonce))
+        }), buildMerklePath(wrongMerklePath), sig, preimage).verify()
+
+        expect(result.success, result.error).to.be.false
+    });
+
+    
```

Comment:
Add an e2e test

- when nonce is odd, Bob should NOT win and not able to unlock